engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/EqualsNode.java

```diff
@@ -133,17 +146,51 @@ boolean equalsDoubleBigInt(double self, EnsoBigInteger other) {
 
   @Specialization
   boolean equalsUnresolvedSymbols(UnresolvedSymbol self, UnresolvedSymbol otherSymbol,
-                                  @Cached EqualsAnyNode equalsNode) {
+                                  @Cached EqualsNode equalsNode) {
     return self.getName().equals(otherSymbol.getName())
         && equalsNode.execute(self.getScope(), otherSymbol.getScope());
   }
 
   @Specialization
   boolean equalsUnresolvedConversion(UnresolvedConversion selfConversion, UnresolvedConversion otherConversion,
-                                     @Cached EqualsAnyNode equalsNode) {
+                                     @Cached EqualsNode equalsNode) {
     return equalsNode.execute(selfConversion.getScope(), otherConversion.getScope());
   }
 
+  @Specialization
+  boolean equalsModuleScopes(ModuleScope selfModuleScope, ModuleScope otherModuleScope,
+      @Cached EqualsNode equalsNode) {
+    return equalsNode.execute(selfModuleScope.getModule(), otherModuleScope.getModule());
+  }
+
+  @Specialization
+  @TruffleBoundary
+  boolean equalsModules(Module selfModule, Module otherModule,
+      @Cached EqualsNode equalsNode) {
+    return equalsNode.execute(selfModule.getName().toString(), otherModule.getName().toString());
+  }
+
+  @Specialization
+  boolean equalsFiles(EnsoFile selfFile, EnsoFile otherFile,
+      @CachedLibrary(limit = "5") InteropLibrary interop) {
+    return equalsStrings(selfFile.getPath(), otherFile.getPath(), interop, interop);
+  }
+
+  /**
+   * There is no specialization for {@link TypesLibrary#hasType(Object)}, because also
+   * primitive values would fall into that specialization and it would be too complicated
+   * to make that specialization disjunctive. So we rather specialize directly for
+   * {@link Type types}.
+   */
+  @Specialization
+  boolean equalsTypes(Type selfType, Type otherType,
+      @Cached EqualsNode equalsNode) {
+    return equalsNode.execute(
```

Comment:
Specialization for `Type` is OK. Two types are equal when `selfType == otherType`. No need to play with names.